dashboard/app/controllers/student_work_sample_controller.rb

```diff
@@ -81,68 +81,22 @@ def fetch_student_code_samples
       return render status: :not_found, json: "Unit with id #{unit_id}"
     end
 
-    if student_work_params[:include_ai_evaluations]
-      fetch_student_code_samples_with_evaluations(level, unit_id, num_samples)
-    else
-      fetch_student_code_samples_without_evaluations(level, unit_id, num_samples)
-    end
-  end
-
-  def fetch_student_code_samples_without_evaluations(level, unit_id, num_samples)
-    # We want to pull samples from students who have been assigned to work on the level.
-    sections = Section.where(script_id: unit_id)
-    student_ids = Follower.where(section: sections).pluck(:student_user_id)
+    # Find users who have worked on this level.
+    student_ids = UserLevel.where(level_id: level.id, script_id: unit_id).pluck(:user_id)
```

Comment:
@kobryan0619 made a great point on an earlier review that because we're looking at historical data the most recently assigned unit might not be the unit we care about; for example if the section completed all of CSP and we want data from CSP unit 4 there will likely be many students assigned to the last unit of CSP at the time we try to fetch the work samples. This widens our sampling pool.